RFP-01.md

```diff
@@ -0,0 +1,169 @@
+# RFP-01: Settlement contract wrapper
+
+## Background
+
+The CoW DAO has recently passed [CIP-44: Reduced bonding requirements](https://snapshot.org/#/cow.eth/proposal/0x1b6f1171633ec3d20c4370db37074aa1bd830486d4d0d6c26165915cc42d9412), which allows solvers to participate in the batch auction with reduced bonding requirements. The reduced bonding requirements are as follows:
+
+- $50k in yield-bearing stable coins or ETH
+- 500k COW
+
+The target deposit is:
+
+- $100k in yield-bearing stable coins or ETH
+- 1M COW
+
+## Goal
+
+The goal of this project is to implement a `SubPoolFactory` that allows solvers to participate in the batch auction with reduced bonding requirements. The `SubPoolFactory` will deploy a `SubPool` for each solver, which will be funded to at least the minimum required amount. The `SubPoolFactory` will also provide a wrapper around the `GPv2Settlement` contract that allows solvers to execute their solutions by providing a signature attesting to the solution and their parameters.
+
+### Key Features
+
+- Ultra gas efficient signed settlement function
+- No co-mingling of funds between solvers
+- Easily observable on-chain addresses for MEV Blocker rebates
+- Permissionless access to the batch auction with reduced bonding requirements
+
+### Deliverables
+
+- Contract system for `SubPoolFactory`, `SubPool`, and `SignedSettlement`
+- Comprehensive documentation
+- Open-source code repository
+
+### Interface
+
+```solidity
+interface Auth {
+    /**
+     * Allows `usr` to perform root-level functions
+     */
+    function rely(address usr) external auth;
+    /**
+     * Denies `usr` from being able to perform root-level functions
+     */
+    function deny(address usr) external auth;
+    /**
+     * Determine if `usr` is able to perform root-level functions
+     */
+    function wards(address usr) external view returns (bool);
+}
+
+interface SubPool is Auth {
```

Comment:
Why does this need to be auth (looks like there is nothing that requires only owner).

I guess we want to allow changes to the backend url?

Reply:
Will need it for announceDestroy too. But we can move all the auth calls to the factory with a limitation that only the owner would be able to call those. With Auth implemented the pool owner can allow other addresses to do those actions.

Reply:
The reason for `auth` is that the way `auth` is designed allows for _multiple_ (what you call) owners. I think root level functionality on sub-pools should _not_ be bound to the specific EOA that is `tx.origin` on the transaction.

Consider the case of a solver who is using a hardware signing key, or a service like AWS KMS for signing. Having to then use those keys to adjust the backend URI, or exit may be onerous. So this method would allow for a subpool to be bound to a specific signing key for submitting the settlement, however also allow for other ways for them to administer their sub pool through whatever method they want.

If this is the case then, having the auth list inside the SubPool is a little nicer IMO than having a multi-dimensional auth mapping in `SubPoolFactory` and allows for a "separation of concerns" with respect to authorisation (if you want to know if address `x` is authorised on the sub pool - you ask the sub pool).

Reply:
I see, but we also wanted a function to determine for a given address what is the pool it belongs to, right (so that it's easy for someone handing out offchain signatures to a solver EOA to check if they are still collateralized)?

Are we doing this with a map inside the factory?

Reply:
> Are we doing this with a map inside the factory?

I'd leave this open to the grantee. This could be implemented using either deterministic addresses, or a mapping. Functions on the factory are `solverSubPool` (to get from solver EOA -> pool address), and `canSolve` which does the view on-chain validation that the solver is collateralised at the level required (and has a valid sub pool etc).

Reply:
Collateralisation and types, as the sub-pools are KYC'ed is now delegated to off-chain handling.